File: tests/human/x/java/Closure.java
import java.util.function.IntUnaryOperator;

public class Closure {
    static IntUnaryOperator makeAdder(int n) {
        return x -> x + n;
    }
    public static void main(String[] args) {
        IntUnaryOperator add10 = makeAdder(10);
        System.out.println(add10.applyAsInt(7));
    }
}

File: tests/human/x/pas/closure.pas
program Closure;

type
  TAdder = function(x: integer): integer;

function makeAdder(n: integer): TAdder;
  function add(x: integer): integer;
  begin
    add := x + n;
  end;
begin
  makeAdder := @add;
end;

var
  add10: TAdder;
begin
  add10 := makeAdder(10);
  Writeln(add10(7));
end.
